### PR TITLE
Hungry feature

### PR DESCRIPTION
**Pull request by RamenMode**
## Proposed Changes

Give an overall description of the changes, along with the context and motivation.
Mention relevant issues and pull requests as needed.

Changes 
- Removed use of hungry_minimum_factor as a metric for how hungry a queue is (still use hungry-minimum)
- m.hungry now returns the additional number of tasks the queue can take if it is hungry and still 0 if not
- Estimation is based on aggregation of current tasks in queue required resources and should be used for "similarly" resourced tasks

## Merge Checklist

The following items must be completed before PRs can be merge.
Check these off to verify you have completed all steps.

- [X] `make test`       Run local tests prior to pushing.
- [X] `make format`     Format source code to comply with lint policies. Note that some lint errors can only be resolved manually (e.g., Python)
- [X] `make lint`       Run lint on source code prior to pushing.
- [X] Manual Update     Update the manual to reflect user-visible changes.
- [X] Type Labels       Select a github label for the type: bugfix, enhancement, etc.
- [X] Product Labels    Select a github label for the product: TaskVine, Makeflow, etc.
- [X] PR RTM            Mark your PR as ready to merge.

### CODE REVIEW

File: taskvine/src/manager/.vscode/c_cpp_properties.json [hunk 1]
```diff
@@ -0,0 +1,16 @@
+{
```
Comment (dthain) [inline taskvine/src/manager/.vscode/c_cpp_properties.json:1]: Don't include this file.

File: taskvine/src/manager/vine_manager.c [hunk 2]
```diff
@@ -5151,20 +5152,23 @@ int vine_hungry(struct vine_manager *q)
 
 	int count = task_state_count(q, NULL, VINE_TASK_READY);
 
-	while (count > 0) {
-		count--;
+	// represents number of tasks in queue
+	int i = count;
+
+	while (i > 0) {
```
Comment (dthain) [inline taskvine/src/manager/vine_manager.c:5158]: Use LIST_ITERATE here.

**Review by dthain — CHANGES_REQUESTED** [review]
Overall, the right idea, see fixes -

File: taskvine/src/manager/vine_manager.c (hunk 2 above)
Comment (RamenMode) [reply taskvine/src/manager/vine_manager.c]: changed

File: taskvine/src/manager/.vscode/c_cpp_properties.json (hunk 1 above)
Comment (RamenMode) [reply taskvine/src/manager/.vscode/c_cpp_properties.json]: changed

**Comment by RamenMode** [comment]
Made the necessary changes

**Review by dthain — APPROVED** [review]

**Pull request merged**